### ISSUE TITLE
Different results from Kipf paper

### ISSUE THREAD

**Issue by Keramatfar** [danielegrattarola/spektral#107]
Thanks for your package,
I just test this sample code: https://github.com/danielegrattarola/spektral/blob/master/examples/node_prediction/citation_gcn.py, and got very different results from Kipf paper. What is the reason?

**Comment by danielegrattarola** [comment]
Hi, 

are we talking orders of magnitude or just a couple % points? 
I get around 80% test accuracy when running the example as-is.

Anyway, sometimes GCN may fail to converge. Try raising the number of epochs by a few hundreds, it should comfortably get at around 81% on Cora.

Cheers

**Comment by Keramatfar** [comment]
You know that, as i think they reported more than 85%. I think they used bag of word representation, but the example just uses one hot encoding for document representation. That maybe explains the difference.

**Comment by danielegrattarola** [comment]
If you mean the GCN paper by Kipf and Welling, they report 81.5% using the fixed Planetoid splits on Cora and 80.1 using random splits. 

In my post above I didn't mention that you should also increase the patience, not just the epochs. Basically, train the network more to account for differences in implementation between tf.keras and the original TF code by Kipf.
You should be able to replicate the results and possibly even do better.

The representation of the node feature should be the same, as the code was taken verbatim from https://github.com/tkipf/gcn

**Comment by Keramatfar** [comment]
I rechecked, you are right.
